snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -433,38 +433,43 @@ def setup_assets(self) -> None:
 
     def _generate_command_chain(self) -> List[str]:
         command_chain = list()
-        # Classic confinement or building on a host that does not match the target base
-        # means we cannot setup an environment that will work.
+
+        # command-chain is not required in these situations.
         if (
-            self._config_data["confinement"] == "classic"
+            self._config_data.get("type", "app") == "snapd"
```

Comment:
I don't think specifying a default for .get() matters.  `None` != `"snapd"`.

Reply:
it does not, but the default behavior for not having it specified is "app", I thought this would make it clear what the outcome is when not set.